### DISCUSSION TITLE
how can I implement my own version of  `auth/forgot-password` ?

### DISCUSSION THREAD

**Discussion by DinaTaklit** [fastapi-users/fastapi-users#886] [Q&A]
Hello, I have tried to run  `auth/forgot-password` but it does nothing, Could reimplement this end point? how?

**Comment by frankie567** [comment]
By default, the endpoint will just retrieve the user and generate a reset password token. It's then up-to-you to implement the logic coming after (typically, sending an e-mail).

To do this, you just need to implement the `on_after_forgot_password` method on `UserManager`. See: https://fastapi-users.github.io/fastapi-users/configuration/user-manager/#on_after_forgot_password
